src/Core/Tool/Authorizer/HXLMetadataAuthorizer.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * Ushahidi HXLMetadataAuthorizer Authorizer
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\Core\Tool\Authorizer;
+
+use Ushahidi\Core\Entity;
+use Ushahidi\Core\Entity\Config;
+use Ushahidi\Core\Entity\User;
+use Ushahidi\Core\Entity\UserRepository;
+use Ushahidi\Core\Entity\Permission;
+use Ushahidi\Core\Tool\Authorizer;
+use Ushahidi\Core\Traits\AdminAccess;
+use Ushahidi\Core\Traits\OwnerAccess;
+use Ushahidi\Core\Traits\UserContext;
+use Ushahidi\Core\Traits\PrivAccess;
+use Ushahidi\Core\Tool\Permissions\AclTrait;
+
+// The `HXLMetadataAuthorizer` class is responsible for access checks on `HXLMetadata` Entity
+class HXLMetadataAuthorizer extends HXLAuthorizer
+{
+    // The access checks are run under the context of a specific user
+    use UserContext;
+
+    // It uses `AdminAccess` to check if the user has admin access
+    use AdminAccess;
+
+    // It uses `PrivAccess` to provide the `getAllowedPrivs` method.
+    use PrivAccess;
+
+    // Check that the user has the necessary permissions
+    // if roles are available for this deployment.
+    use AclTrait;
+
+    use OwnerAccess;
+
+
+    /* Authorizer */
+    public function isAllowed(Entity $entity, $privilege)
+    {
+        if (parent::isAllowed($entity, $privilege)) {
```

Comment:
It'd be cleaner to express this as `parent::isAllowed() && $this->isUserOwner()`.

However if we're going to use the HXL Authorizer everywhere like this, it might be better as a DataImportExport trait that provides `$this->canImportExportData()`.

The original ideal of authorizers was that you could read them and understand the intent. So a parent::isAllowed() check doesn't provide much info about intent here.

Reply:
I'll also caveat this with: if you'd rather just get on with it and keep coding, this isn't a blocking/must fix request.